add docs for equations (#289)
CliMA/ClimaAtmos.jl: docs/src/equations.md

  - ``\rho e_\text{int}``: _internal energy_ in J/m³
* ``\rho \theta``: _other conserved scalars_ (moisture, tracers, etc), again stored at cell centers.

## Operators

We make use of the following operators

### Reconstruction

* ``I^c`` is the [face-to-center reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateF2C)
* ``I^f`` is the [center-to-face reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateC2F)
  - Currently this is just the arithmetic mean, but we will need to use a weighted version with stretched vertical grids.
* ``U^f`` is the [center-to-face upwind product operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)
  - Currently this is first-order, we will switch this to 3rd order.

### Differentiation operators

- ``D_h`` is the [discrete horizontal spectral divergence](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Divergence)
- ``D^c_v`` is the [face-to-center vertical divergence](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.DivergenceF2C)

>   - The fluxes at the surface and top are set to zero.

Reviewer: ```suggestion
  - The fluxes are set to zero at the top and bottom boundaries.
```

- ``G_h`` is the [discrete horizontal spectral gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Gradient)
- ``G^f_v`` is the [center-to-face vertical gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.GradientC2F)

>   - the gradient is set to 0 at the boundary.

Reviewer: ```suggestion
  - the gradient is set to 0 at the top and bottom boundaries.
```

- ``C_h`` is the [horizontal curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Curl)
  - ``C_h[\boldsymbol{u}_h]`` returns a vector with only a vertical _contravariant_ component.
  - ``C_h[\boldsymbol{u}_v]`` returns a vector with only a horizontal _contravariant_ component.
- ``C^f_v`` is the [center-to-face vertical curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.CurlC2F)
  - ``C^f_v[\boldsymbol{u}_h]`` returns a vector with only a horizontal _contravariant_ component.

>   - the boundary is defined to be zero.

Reviewer: ```suggestion
  - the curl is set to 0 at the top and bottom boundaries.
```

# Equations

!!! note

    This follows what is _currently_ implemented in `examples`: it should be kept up-to-date as code is modified. If you think something _should_ be changed (but hasn't been), please add a note.

## Prognostic variables

* ``\rho``: _mass_ in kg/m³. This is discretized at cell centers.

> * ``\boldsymbol{u}`` _velocity_, a vector in m/s. This is discretized via ``\boldsymbol{u} = \boldsymbol{u}_v + \boldsymbol{u}_h`` where

Reviewer: ```suggestion
* ``\boldsymbol{u}`` _velocity_, a vector in m/s. This is discretized via ``\boldsymbol{u} = \boldsymbol{u}_h + \boldsymbol{u}_v`` where
```

  - ``\boldsymbol{u}_h = u_1 \boldsymbol{e}^1 + u_2 \boldsymbol{e}^2`` are the horizontal covariant part (with respect to the reference element), stored at cell centers.
  - ``\boldsymbol{u}_v = u_3 \boldsymbol{e}^3`` is the vertical covariant part, stored at cell faces.
* _energy_, stored at cell centers; can be either:
  - ``\rho e``: _total energy_ in J/m^3
  - ``\rho e_\text{int}``: _internal energy_ in J/m^3
* ``\rho \theta``: _other conserved scalars_ (moisture, tracers, etc), again stored at cell centers.

## Operators

We make use of the following operators

### Reconstruction

* ``I^c`` is the [face-to-center reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateF2C)
* ``I^f`` is the [center-to-face reconstruction operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.InterpolateC2F)
  - Currently this is just the arithmetic mean, but we will need to use a weighted version with stretched vertical grids.
* ``U^f`` is the [center-to-face upwind product operator](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)

>   - Currently this is first-order, we will switch this to 3rd order.

Reviewer: ```suggestion
  - Currently this is first-order, we will switch this to third-order.
```
(Sorry about my OCD here, you can ignore it:)

- ``D^c_v`` is the [face-to-center vertical divergence](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.DivergenceF2C)
  - The fluxes at the surface and top are set to zero.
- ``G_h`` is the [discrete horizontal spectral gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Gradient)
- ``G^f_v`` is the [center-to-face vertical gradient](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.GradientC2F)
  - the gradient is set to 0 at the boundary.
- ``C_h`` is the [horizontal curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.Curl)
  - ``C_h[\boldsymbol{u}_h]`` returns a vector with only a vertical _contravariant_ component.
  - ``C_h[\boldsymbol{u}_v]`` returns a vector with only a horizontal _contravariant_ component.
- ``C^f_v`` is the [center-to-face vertical curl](https://clima.github.io/ClimaCore.jl/stable/operators/#ClimaCore.Operators.CurlC2F)
  - ``C^f_v[\boldsymbol{u}_h]`` returns a vector with only a horizontal _contravariant_ component.
  - the boundary is defined to be zero.

## Auxiliary and derived quantities

> * ``\boldsymbol{\Omega}`` is the planetary angular velocity (Coriolis). We currently use a shallow-atmosphere approximation, with

Reviewer: ```suggestion
* ``\boldsymbol{\Omega}`` is the planetary angular velocity. We currently use a shallow-atmosphere approximation, with
```
